### PR TITLE
refactor: Add AssetFolder as AugmentedPath

### PR DESCRIPTION
**Pull request by seakayone**
Introduce more `AugmentedPath` types:
* `AssetsBaseFolder` - the "images" folder where all assets reside
* `TempFolder` - dsp-ingest's own temp folder
* `AssetFolder` - a folder in which an asset is in

Add conversion from `AugmentedPath` to `Path`.

Add methods from `Path` to `AugementedPath`:
```
  def /(other: Path): Path
  def /(other: String): Path 
  def toAbsolutePath: IO[IOError, Path]
  def parent: Option[Path] 
  def toFile: java.io.File 
  ```

### CODE REVIEW

File: src/main/scala/swiss/dasch/domain/AugmentedPath.scala [hunk 1]
```diff
@@ -22,8 +25,15 @@ trait AugmentedPathBuilder[A <: AugmentedPath] {
 
 trait AugmentedPath {
   def path: Path
+  def /(other: Path): Path              = path / other
+  def /(other: String): Path            = path / other
+  def toAbsolutePath: IO[IOError, Path] = path.toAbsolutePath
+  def parent: Option[Path]              = path.parent
+  def toFile: java.io.File              = path.toFile
```
Comment (BalduinLandolt) [inline src/main/scala/swiss/dasch/domain/AugmentedPath.scala:32]:
```suggestion
  final def /(other: Path): Path              = path / other
  final def /(other: String): Path            = path / other
  final def toAbsolutePath: IO[IOError, Path] = path.toAbsolutePath
  final def parent: Option[Path]              = path.parent
  final def toFile: java.io.File              = path.toFile
```

File: src/main/scala/swiss/dasch/domain/StorageService.scala [hunk 2]
```diff
@@ -41,23 +42,28 @@ trait StorageService {
...
    */
   def saveJsonFile[A](file: Path, content: A)(implicit encoder: JsonEncoder[A]): Task[Unit]
+
   def copyFile(source: Path, target: Path, copyOption: CopyOption*): IO[IOException, Unit]
+
   def createDirectories(path: Path, attrs: FileAttribute[_]*): IO[IOException, Unit]
 
   /**
-   * Deletes a file.
+   * Deletes a a regular file.
```
Comment (BalduinLandolt) [inline src/main/scala/swiss/dasch/domain/StorageService.scala:51]:
```suggestion
   * Deletes a regular file.
```

**Review by BalduinLandolt — APPROVED** [review]
LGTM, I added two small suggestions

File: src/main/scala/swiss/dasch/domain/StorageService.scala [hunk 3]
```diff
@@ -22,10 +23,10 @@ import java.time.format.DateTimeFormatter
...
 
 trait StorageService {
-  def getProjectDirectory(projectShortcode: ProjectShortcode): UIO[ProjectFolder]
-  def getAssetDirectory(asset: AssetRef): UIO[Path]
-  def getAssetDirectory(): UIO[Path]
-  def getTempDirectory(): UIO[Path]
+  def getProjectFolder(projectShortcode: ProjectShortcode): UIO[ProjectFolder]
+  def getAssetFolder(asset: AssetRef): UIO[AssetFolder]
+  def getAssetsBaseFolder(): UIO[AssetsBaseFolder]
+  def getTempFolder(): UIO[TempFolder]
```
Comment (mpro7) [inline src/main/scala/swiss/dasch/domain/StorageService.scala:29]: What is the motivation behind that rename? I can still see some "dir" / "directory" props floating around.

**Review by mpro7 — APPROVED** [review]
Just one comment below:

File: src/main/scala/swiss/dasch/domain/StorageService.scala (hunk 3 above)
Comment (seakayone) [reply src/main/scala/swiss/dasch/domain/StorageService.scala:29]: Aligning the method names with the returned types.

**Pull request merged**